xla/backends/gpu/runtime/thunk.h

```diff
@@ -356,26 +362,53 @@ class Thunk {
 
     int64_t execution_id = 0;
 
+    mutable CommandBufferParams* record_params = nullptr;
+
+    // RAII helper to temporarily change record_params and automatically
+    // restore it when the scope exits.
+    class ScopedCommandBufferParams {
```

Comment:
Instead of this hack with `mutable` it should be possible to add something like:


```
ExecuteParams ExecuteParams::CopyWith(CommandBufferParams*);
```

all members are cheap to copy and it's less confusing